azure-devops/azext_devops/dev/pipelines/github_api_helper.py

```diff
@@ -0,0 +1,146 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import requests
+from knack.log import get_logger
+from knack.util import CLIError
+from knack.prompting import prompt
+from azext_devops.dev.common.git import resolve_git_ref_heads, get_branch_name_from_ref
+
+logger = get_logger(__name__)
+
+_HTTP_NOT_FOUND_STATUS = 404
+_HTTP_SUCCESS_STATUS = 200
+
+class Files:  # pylint: disable=too-few-public-methods
+    def __init__(self, path, content):
+        self.path = path
+        self.content = content
+
+
+def get_github_pat_token():
+    from azext_devops.dev.common.github_credential_manager import GithubCredentialManager
+    github_manager = GithubCredentialManager()
+    return github_manager.get_token()
+
+
+def get_github_repos_api_url(repo_id):
+    return 'https://api.github.com/repos/' + repo_id
+
+
+def push_files_github(files, repo_name, branch, commit_to_branch, message="Set up CI with Azure Pipelines"):
+    if commit_to_branch:
+        commit_files_to_github_branch(files, repo_name, branch, message)
+    else:
+        # Create Branch
+        new_branch = create_github_branch(repo=repo_name, source=branch)
+        # Commit files to branch
+        commit_files_to_github_branch(files, repo_name, new_branch, message)
+        # Create PR from new branch
+        pr = create_pr_github(branch, new_branch, repo_name, message)
+        print('Created a Pull Request - {url}'.format(url=pr['url']))
+
+
+def create_pr_github(branch, new_branch, repo_name, message):
+    token = get_github_pat_token()
+    create_pr_url = 'https://api.github.com/repos/{repo_id}/pulls'.format(repo_id=repo_name)
+    create_pr_request_body = {
+        "title": message,
+        "head": new_branch,
+        "base": branch
+        }
+    create_response = requests.post(url=create_pr_url, auth=('', token),
+                                    json=create_pr_request_body, headers=get_application_json_header())
+    if not create_response.status_code == 201:
+        raise CLIError('Pull request creation failed. Error: ({err})'.format(err=create_response.reason))
+    import json
+    return json.loads(create_response.text)
+
+
+def create_github_branch(repo, source):
+    token = get_github_pat_token()
+    # Validate new branch name is valid
+    branch_is_valid = False
+    while not branch_is_valid:
+        new_branch = prompt(msg='Enter new branch name to create: ')
+        if not get_github_branch(repo, new_branch):
+            branch_is_valid = True
+        else:
+            logger.warning('Not a valid branch name.')
```

Comment:
do we want to show this info to user? I think logger will not show up and we should use prompt..
**not sure**

Reply:
Warning shows on the console.